crates/red_knot_python_semantic/src/semantic_index/definition.rs

```diff
@@ -39,6 +39,7 @@ impl<'db> Definition<'db> {
 pub(crate) enum DefinitionNodeRef<'a> {
     Import(&'a ast::Alias),
     ImportFrom(ImportFromDefinitionNodeRef<'a>),
+    For(&'a ast::StmtFor),
```

Comment:
I think we might want to define a custom struct which determines which variable this definition is for. Considering the below example:
```py
for k, v in items:
	...
```
We will need to create two definitions for `k` and `v` variable and this new struct would capture that. It should probably be similar to `AssignmentDefinitionNodeRef` (similarly `AssignmentDefinitionNodeKind`), so:
```rs
pub(crate) struct ForStmtDefinitionNodeRef<'a> {
	pub(crate) node: &'a ast::StmtFor,
	pub(crate) target: &'a ast::ExprName,
}
```

Maybe we could get away by just storing the `iter` instead of the entire node?